extensions/wikia/AdEngine/js/provider/gpt/googleTag.js

```diff
@@ -0,0 +1,149 @@
+/*global define,setTimeout*/
+/*jshint maxlen:125, camelcase:false, maxdepth:7*/
+define('ext.wikia.adEngine.provider.gpt.googleTag', [
+	'wikia.document',
+	'wikia.log',
+	'wikia.window'
+], function (doc, log, window) {
+
+	var logGroup = 'ext.wikia.adEngine.provider.gpt.googleTag',
+		initialized = false,
+		registeredCallbacks = {},
+		slots = {},
+		slotQueue = [],
+		pageLevelParams,
+		googleTag,
+		pubAds;
+
+	function init() {
+		log('init', 'debug', logGroup);
+
+		var gads = doc.createElement('script'),
+			node = doc.getElementsByTagName('script')[0];
+
+		window.googletag = window.googletag || {};
+		window.googletag.cmd = window.googletag.cmd || [];
+
+		gads.async = true;
+		gads.type = 'text/javascript';
+		gads.src = '//www.googletagservices.com/tag/js/gpt.js';
+
+		log('Appending GPT script to head', 'debug', logGroup);
+
+		node.parentNode.insertBefore(gads, node);
+		googleTag = window.googletag;
+
+		// Enable services
+		log(['init', 'googletag.cmd.push'], 'info', logGroup);
+		googleTag.cmd.push(function () {
+			pubAds = googleTag.pubads();
+
+			pubAds.collapseEmptyDivs();
+			pubAds.enableSingleRequest();
+			pubAds.disableInitialLoad(); // manually request ads using refresh
+			pubAds.addEventListener('slotRenderEnded', dispatchEvent);
+
+			googleTag.enableServices();
+
+			log(['init', 'googletag.cmd.push', 'done'], 'debug', logGroup);
+		});
+		initialized = true;
+	}
+
+	function isInitialized() {
+		return initialized;
+	}
+
+	function setPageLevelParams(params) {
+		var name,
+			value;
+
+		googleTag.cmd.push(function () {
+			pageLevelParams = params;
```

Comment:
`pageLevelParams` variable is used later to configure data-page-level-params on single slot while adding